assembly/src/assembler.rs

```diff
@@ -140,6 +143,15 @@ impl Assembler {
     }
 }
 
+// ------------------------------------------------------------------------------------------------
+/// Accessors
+impl Assembler {
+    /// Access to the inner [`SourceManager`] implementation.
+    pub fn source_manager(&self) -> Arc<dyn SourceManagerSync> {
+        self.source_manager.clone()
+    }
+}
```

Comment:
I don't think we should add this method to the assembler (we had it before and removed recently because we didn't want the assembler to assume that the source manager would be sync). We could refactor the assembler to take source manager as a generic parameter - but I'm not sure that's needed for this PR.

Reply:
+1, definitely don't want to add this back IMO.

Reply:
@bitwalker I didn't see your comments for some reason on this PR